internal/db/filter/query_builder.go

```diff
@@ -386,7 +386,21 @@ func (qb *QueryBuilder) buildOperatorCondition(column string, operator string, v
 		return conditionResult{condition: fmt.Sprintf("%s LIKE ?", column), args: []any{value}}
 	case "IN":
 		// Handle IN operator with array values
-		return conditionResult{condition: fmt.Sprintf("%s IN ?", column), args: []any{value}}
+		if valueSlice, ok := value.([]interface{}); ok {
+			if len(valueSlice) == 0 {
+				// Empty list should return false condition
+				return conditionResult{condition: "1 = 0", args: []any{}}
+			}
+			// Create placeholders for each value
+			placeholders := make([]string, len(valueSlice))
+			for i := range placeholders {
+				placeholders[i] = "?"
+			}
+			condition := fmt.Sprintf("%s IN (%s)", column, strings.Join(placeholders, ","))
```

Comment:
Do what you want, of course, but here's another option:

```suggestion
			// Create placeholders for each value
			condition := fmt.Sprintf("%s IN (?%s)", column, strings.Repeat(",?", len(valueSlice)-1))
```